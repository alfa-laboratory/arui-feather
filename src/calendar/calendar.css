/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

@import '../vars.css';

.calendar {
    display: inline-block;
    padding: 0;
    outline: 0;
    font-family: var(--font);
    font-size: var(--font-size-m);
    font-weight: var(--font-weight-normal);
    border-radius: var(--border-radius-control);

    &, *, *:before, *:after {
        box-sizing: border-box;
    }

    &__name {
        text-align: center;
    }

    &__layout {
        border-collapse: separate;
        border-spacing: 0;
    }

    &__title {
        vertical-align: middle;
        padding: 20px 20px 7px;
        line-height: 20px;
    }

    &__layout {
        padding: 0 10px 10px;
    }

    &__dayname,
    &__day {
        font-size: var(--font-size-s);
    }

    &__dayname {
        cursor: default;
        font-size: var(--font-size-xs);
        font-weight: var(--font-weight-normal);
        line-height: 34px;
        text-align: center;
    }

    &__day {
        width: 35px;
        cursor: pointer;
        line-height: 30px;
        text-align: center;
    }

    &__arrow, &__double_arrow {
        opacity: var(--opacity-minor);
        width: 18px;
        height: 18px;
        cursor: pointer;
        transition: all 0.1s ease-out;
        background-position: center;
        background-repeat: no-repeat;
    }

    &__arrow:hover {
        opacity: var(--opacity-active);
    }

    &__arrow_direction_left, &__double_arrow_direction_left {
        float: left;
    }

    &__arrow_direction_right, &__double_arrow_direction_right {
        float: right;
    }

    &__day_type_weekend {
        font-weight: var(--font-weight-medium);
    }

    &__day_state_current {
        position: relative;
        margin: 0;
        outline: 0;
        user-select: none;
        color: var(--color-accent-content);
        text-align: center;
        z-index: 2;

        &:before {
            content: '';
            position: absolute;
            top: 3px;
            right: 0;
            bottom: 2px;
            left: 0;
            background: var(--color-accent);
            border-radius: 14px;
            z-index: -1;
        }
    }

    &__day_type_off,
    &__day_type_weekend-off {
        cursor: default;
        opacity: var(--opacity-minor);
    }

    &__arrow_disabled {
        visibility: hidden;
    }
}
