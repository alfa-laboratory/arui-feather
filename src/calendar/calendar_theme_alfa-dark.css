@import "@alfalab/core-components-vars/colors-bluetint.css";

.calendar_theme_alfa-dark {
    color: var(--color-dark-text-primary);
    background-color: var(--color-dark-bg-secondary);

    .calendar {
        &__event {
            background-color: var(--color-dark-bg-secondary);

            &:after {
                background-color: var(--color-dark-bg-secondary-inverted);
            }
        }

        &__wrapper {
            background-color: var(--color-dark-bg-secondary);

            &:before {
                background: linear-gradient(
                    to bottom,
                    var(--color-dark-bg-secondary) 0%,
                    var(--color-dark-bg-secondary) 33%,
                    rgba(255, 255, 255, 0) 100%
                );
            }
        }

        &__day {
            &:hover {
                background: var(--color-dark-bg-tertiary);
            }

            &_state_today {
                box-shadow: inset 0 0 0 1px
                    var(--color-dark-border-secondary-inverted);
            }

            &_state_current {
                color: var(--color-dark-text-primary);
                background: var(--color-dark-bg-primary);

                &:hover {
                    background: var(--color-dark-bg-primary);
                }
            }
        }

        &__dayname {
            color: var(--color-dark-text-secondary);
        }

        &__day_empty,
        &__select_type_off,
        &__day_type_off,
        &__day_type_weekend-off {
            color: var(--color-dark-text-tertiary);

            &:hover {
                background-color: transparent;
            }
        }

        &__name {
            &_month,
            &_year {
                background-color: var(--color-dark-bg-tertiary);

                &:hover {
                    background-color: var(--color-dark-bg-primary);
                }
            }
        }

        &__select {
            background: var(--color-dark-bg-tertiary);
            border: 1px solid var(--color-dark-border-tertiary);

            &:hover {
                background: var(--color-dark-bg-primary);
                border-color: var(--color-dark-border-primary);
            }

            &_type_off:hover {
                background: var(--color-dark-bg-tertiary);
                border-color: var(--color-dark-border-tertiary);
            }

            &_state_current {
                border-color: var(--color-dark-border-secondary-inverted);

                &:hover {
                    border-color: var(--color-dark-border-secondary-inverted);
                }
            }
        }

        &__select-arrows {
            background-image: url(./images/calendar__select-arrows-color.svg);
        }

        &__arrow_direction_left {
            background-image: url(./images/calendar__arrow-left-color.svg);
        }

        &__arrow_direction_right {
            background-image: url(./images/calendar__arrow-right-color.svg);
        }
    }
}
