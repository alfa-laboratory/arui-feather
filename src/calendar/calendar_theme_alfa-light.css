@import "@alfalab/core-components-vars/colors-bluetint.css";

.calendar_theme_alfa-light {
    color: var(--color-light-text-primary);
    background-color: var(--color-light-bg-secondary);

    .calendar {
        &__event {
            background-color: var(--color-light-bg-secondary);

            &:after {
                background-color: var(--color-light-bg-secondary-inverted);
            }
        }

        &__wrapper {
            background-color: var(--color-light-bg-secondary);

            &:before {
                background: linear-gradient(
                    to bottom,
                    var(--color-light-bg-secondary) 0%,
                    var(--color-light-bg-secondary) 33%,
                    rgba(255, 255, 255, 0) 100%
                );
            }
        }

        &__day {
            &:hover {
                background: var(--color-light-bg-tertiary);
            }

            &_state_today {
                box-shadow: inset 0 0 0 1px
                    var(--color-light-border-secondary-inverted);
            }

            &_state_current {
                color: var(--color-light-text-primary);
                background: var(--color-light-bg-primary);

                &:hover {
                    background: var(--color-light-bg-primary);
                }
            }
        }

        &__dayname {
            color: var(--color-light-text-secondary);
        }

        &__day_empty,
        &__select_type_off,
        &__day_type_off,
        &__day_type_weekend-off {
            color: var(--color-light-text-tertiary);

            &:hover {
                background-color: transparent;
            }
        }

        &__name {
            &_month,
            &_year {
                background-color: var(--color-light-bg-tertiary);

                &:hover {
                    background-color: var(--color-light-bg-primary);
                }
            }
        }

        &__select {
            background: var(--color-light-bg-tertiary);
            border: 1px solid var(--color-light-border-tertiary);

            &:hover {
                background: var(--color-light-bg-primary);
                border-color: var(--color-light-border-primary);
            }

            &_type_off:hover {
                background: var(--color-light-bg-tertiary);
                border-color: var(--color-light-border-tertiary);
            }

            &_state_current {
                border-color: var(--color-light-border-secondary-inverted);

                &:hover {
                    border-color: var(--color-light-border-secondary-inverted);
                }
            }
        }

        &__select-arrows {
            background-image: url(./images/calendar__select-arrows-white.svg);
        }

        &__arrow_direction_left {
            background-image: url(./images/calendar__arrow-left-white.svg);
        }

        &__arrow_direction_right {
            background-image: url(./images/calendar__arrow-right-white.svg);
        }
    }
}
