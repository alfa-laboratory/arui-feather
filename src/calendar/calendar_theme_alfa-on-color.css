@import '../vars.css';

.calendar_theme_alfa-on-color {
    color: var(--color-content-alfa-on-color);
    background-color: var(--color-background-alfa-on-color);

    .calendar__arrow_direction_left {
        background-image: url('./images/arrow_cl.svg');

        &.calendar__arrow_double {
            background-image: url('./images/2_arrow_cl.svg');
        }
    }

    .calendar__arrow_direction_right {
        background-image: url('./images/arrow_cr.svg');

        &.calendar__arrow_double {
            background-image: url('./images/2_arrow_cr.svg');
        }
    }
}
