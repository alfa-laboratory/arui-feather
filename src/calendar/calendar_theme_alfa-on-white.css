@import '../vars.css';

.calendar_theme_alfa-on-white {
    color: var(--color-content-alfa-on-white);
    background-color: var(--color-background-alfa-on-white);

    .calendar__arrow_direction_left {
        background-image: url('./images/arrow_wl.svg');

        &.calendar__arrow_double {
            background-image: url('./images/2_arrow_wl.svg');
        }
    }

    .calendar__arrow_direction_right {
        background-image: url('./images/arrow_wr.svg');

        &.calendar__arrow_double {
            background-image: url('./images/2_arrow_wr.svg');
        }
    }
}
