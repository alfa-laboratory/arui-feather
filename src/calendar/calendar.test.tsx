/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { mount } from 'enzyme';
import timezoneMock from 'timezone-mock';

import addDays from 'date-fns/add_days';
import startOfDay from 'date-fns/start_of_day';
import subtractDays from 'date-fns/sub_days';
import subtractMonth from 'date-fns/sub_months';
import addMonth from 'date-fns/add_months';
import formatDate from 'date-fns/format';

import { Calendar } from './calendar';

import keyboardCode from '../lib/keyboard-code';

// initialize this later, after we will register timezoneMock
let INITIAL_DAY;
let TODAY_DAY;
const DATE_FORMAT = 'DD.MM.YYYY';

describe('calendar', () => {
    beforeAll(() => {
        timezoneMock.register('UTC');
        INITIAL_DAY = startOfDay(new Date('2016-01-15'));
        TODAY_DAY = startOfDay(new Date());
    });

    afterAll(() => {
        timezoneMock.unregister();
    });

    it('should render without problems', () => {
        const calendar = mount(<Calendar month={ new Date('2018-06-15').valueOf() } />);

        expect(calendar).toMatchSnapshot();
    });

    it('should call `onDayClick` callback on day change', () => {
        const onValueChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                onValueChange={ onValueChange }
            />
        );
        const dateToClick = addDays(INITIAL_DAY, 1);
        const dayNodes = calendar
            .find('.calendar__day')
            .findWhere(e => e.length > 0 && e.props()['data-day'] === dateToClick.valueOf());

        dayNodes.simulate('click');

        expect(onValueChange).toHaveBeenCalledWith(dateToClick.valueOf(), '16.01.2016', false);
    });

    it('should display current date', () => {
        const calendar = mount(<Calendar value={ INITIAL_DAY.valueOf() } />);
        const currentDayNode = calendar.find('.calendar__day_state_current');

        expect(currentDayNode.text()).toContain(INITIAL_DAY.getDate().toString());
    });

    it('should display today', () => {
        const calendar = mount(
            <Calendar
                showToday={ true }
            />
        );
        const todayNode = calendar.find('.calendar__day_state_today');

        expect(todayNode.text()).toContain(TODAY_DAY.getDate().toString());
    });

    it('should display days off', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                offDays={ [addDays(INITIAL_DAY, 1).valueOf()] }
            />
        );
        let dayOffNode = calendar.find('.calendar__day_type_off');

        if (dayOffNode.length === 0) {
            dayOffNode = calendar.find('.calendar__day_type_weekend-off');
        }

        expect(dayOffNode.text()).toContain(`${addDays(INITIAL_DAY, 1).getDate()}`);
    });

    it('should display event days', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                eventDays={ [addDays(INITIAL_DAY, 1).valueOf()] }
            />
        );
        const dayOfEventsNode = calendar.find('.calendar__day_event');

        expect(dayOfEventsNode.text()).toContain(`${addDays(INITIAL_DAY, 1).getDate()}`);
    });

    it('should display earlier limit', () => {
        const EARLIER_LIMIT = 4;
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                earlierLimit={ subtractDays(INITIAL_DAY, EARLIER_LIMIT).valueOf() }
            />
        );
        const daysOffCount = calendar.find('.calendar__day_type_off').length;
        const weekendDaysOffCount = calendar.find('.calendar__day_type_weekend-off').length;

        expect(daysOffCount + weekendDaysOffCount).toEqual(INITIAL_DAY.getDate() - (EARLIER_LIMIT + 1));
    });

    it('should display later limit', () => {
        const LATER_LIMIT = 1;
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                earlierLimit={ addDays(INITIAL_DAY, LATER_LIMIT).valueOf() }
            />
        );
        const daysCount = (new Date(INITIAL_DAY.getFullYear(), INITIAL_DAY.getMonth() + 1, 0)).getDate();
        const daysOffCount = calendar.find('.calendar__day_type_off').length;
        const weekendDaysOffCount = calendar.find('.calendar__day_type_weekend-off').length;

        expect(daysCount - (INITIAL_DAY.getDate() + LATER_LIMIT)).toEqual(daysOffCount + weekendDaysOffCount);
    });

    it('should re set current day if it is day off with earlier limit', () => {
        const EARLIER_LIMIT = 2;
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                earlierLimit={ addDays(INITIAL_DAY, EARLIER_LIMIT).valueOf() }
            />
        );
        const currentDay = calendar.find('.calendar__day_state_current');
        const expectedDay = addDays(INITIAL_DAY, EARLIER_LIMIT).valueOf();

        expect(currentDay.props()['data-day']).toEqual(expectedDay);
    });

    it('should re set current day if it is day off with later limit', () => {
        const LATER_LIMIT = 2;
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                laterLimit={ subtractDays(INITIAL_DAY, LATER_LIMIT).valueOf() }
            />
        );
        const currentDay = calendar.find('.calendar__day_state_current');
        const expectedDay = subtractDays(INITIAL_DAY, LATER_LIMIT).valueOf();

        expect(currentDay.props()['data-day']).toEqual(expectedDay);
    });

    it('should call `onMonthChange` callback on month change', () => {
        const onMonthChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                onMonthChange={ onMonthChange }
            />
        );
        const arrowNode = calendar.find('.calendar__arrow_direction_left');

        arrowNode.at(0).simulate('click');

        expect(onMonthChange).toHaveBeenCalledWith((new Date(2015, 11, 1)).valueOf());
    });

    it('should call `onMonthChange` callback when month was changed by arrow key press', () => {
        const LAST_DAY_OF_MONTH = startOfDay(new Date('2016-01-31'));
        const onMonthChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ LAST_DAY_OF_MONTH.valueOf() }
                onMonthChange={ onMonthChange }
            />
        );

        calendar.simulate('keyDown', { which: keyboardCode.DOWN_ARROW });

        expect(onMonthChange).toHaveBeenCalledWith((new Date(2016, 1, 7)).valueOf());
    });

    it('should select date on next week after down arrow key was pressed', () => {
        const onValueChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                onValueChange={ onValueChange }
            />
        );
        const expectedDate = addDays(INITIAL_DAY, 7);

        calendar.simulate('keyDown', { which: keyboardCode.DOWN_ARROW });

        expect(onValueChange)
            .toHaveBeenCalledWith(expectedDate.valueOf(), formatDate(expectedDate, DATE_FORMAT), true);
    });

    it('should select date on previous week after up arrow key was pressed', () => {
        const onValueChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                onValueChange={ onValueChange }
            />
        );
        const expectedDate = subtractDays(INITIAL_DAY, 7);

        calendar.simulate('keyDown', { which: keyboardCode.UP_ARROW });

        expect(onValueChange)
            .toHaveBeenCalledWith(expectedDate.valueOf(), formatDate(expectedDate, DATE_FORMAT), true);
    });

    it('should select date jump over off day when navigate using keyboard', () => {
        const onValueChange = jest.fn();
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                offDays={ [addDays(INITIAL_DAY, 1).valueOf()] }
                onValueChange={ onValueChange }
            />
        );

        calendar.simulate('keyDown', { which: keyboardCode.RIGHT_ARROW });

        const expectedDate = addDays(INITIAL_DAY, 2);

        expect(onValueChange)
            .toHaveBeenCalledWith(expectedDate.valueOf(), formatDate(expectedDate, DATE_FORMAT), true);
    });

    it('should hide prev month arrows when earlier limit is in current month', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                earlierLimit={ subtractDays(INITIAL_DAY, 10).valueOf() }
            />
        );

        const prevMonthArrow = calendar.find('.calendar__arrow_direction_left');

        expect(prevMonthArrow.getDOMNode().className).toContain('calendar__arrow_disabled');
    });

    it('should hide next month arrows when laterLimit is in current month', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                laterLimit={ addDays(INITIAL_DAY, 10).valueOf() }
            />
        );

        const nextMonthArrow = calendar.find('.calendar__arrow_direction_right');

        expect(nextMonthArrow.getDOMNode().className).toContain('calendar__arrow_disabled');
    });

    it('should show prev month arrow when earlierLimit is in prev month', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                earlierLimit={ subtractMonth(INITIAL_DAY, 1).valueOf() }
            />
        );

        const prevMonthArrow = calendar.find('.calendar__arrow_direction_left').not('.calendar__arrow_double');

        expect(prevMonthArrow.getDOMNode().className).not.toContain('calendar__arrow_disabled');
    });

    it('should show next month arrow when laterLimit is in next month', () => {
        const calendar = mount(
            <Calendar
                value={ INITIAL_DAY.valueOf() }
                laterLimit={ addMonth(INITIAL_DAY, 1).valueOf() }
            />
        );

        const nextMonthArrow = calendar.find('.calendar__arrow_direction_right').not('.calendar__arrow_double');

        expect(nextMonthArrow.getDOMNode().className).not.toContain('calendar__arrow_disabled');
    });

    it('should return root `HTMLElement` after `getNode` method call', () => {
        const calendar = mount<Calendar>(<Calendar />);

        const node = calendar.instance().getNode();

        expect(node).toBeInstanceOf(HTMLElement);
    });

    it('should change month when select one from selection list', () => {
        const calendar = mount(<Calendar />);

        const monthNode = calendar.find('.calendar__name_month');

        monthNode.simulate('click');

        const calendarSelectNode = calendar.find('.calendar__select');

        calendarSelectNode.at(1).simulate('click');

        expect(monthNode.text()).toContain(calendarSelectNode.at(1).text());
    });

    it('should change year when select one from selection list', () => {
        const calendar = mount(<Calendar />);

        const yearNode = calendar.find('.calendar__name_year');

        yearNode.simulate('click');

        const calendarSelectNode = calendar.find('.calendar__select');

        calendarSelectNode.at(1).simulate('click');

        expect(yearNode.text()).toContain(calendarSelectNode.at(1).text());
    });

    it('should make list of years whith amount by default', () => {
        const calendar = mount(<Calendar />);

        const yearNode = calendar.find('.calendar__name_year');

        yearNode.simulate('click');

        const calendarSelectNode = calendar.find('.calendar__select');

        expect(calendarSelectNode.length).toBe(102);
    });

    it('should make correct list of years with established date limits', () => {
        const calendar = mount(<Calendar
            earlierLimit={ 666565200000 }
            laterLimit={ 1582923600000 }
        />);

        const yearNode = calendar.find('.calendar__name_year');

        yearNode.simulate('click');

        const calendarSelectNode = calendar.find('.calendar__select');

        expect(calendarSelectNode.length).toBe(30);
    });
});
