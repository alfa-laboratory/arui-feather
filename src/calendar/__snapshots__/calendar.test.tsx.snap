// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar should render without problems 1`] = `
<Calendar
  eventDays={Array []}
  isKeyboard={true}
  month={1529020800000}
  months={
    Array [
      "Январь",
      "Февраль",
      "Март",
      "Апрель",
      "Май",
      "Июнь",
      "Июль",
      "Август",
      "Сентябрь",
      "Октябрь",
      "Ноябрь",
      "Декабрь",
    ]
  }
  offDays={Array []}
  outputFormat="DD.MM.YYYY"
  selectedFrom={null}
  selectedTo={null}
  showArrows={true}
  showToday={false}
  weekdays={
    Array [
      "Пн",
      "Вт",
      "Ср",
      "Чт",
      "Пт",
      "Сб",
      "Вс",
    ]
  }
>
  <div
    className="calendar"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="grid"
    tabIndex={0}
  >
    <div
      className="calendar__title"
    >
      <div
        className="calendar__arrow calendar__arrow_direction_left"
        data-disabled={false}
        data-step="-1"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      />
      <div
        className="calendar__arrow calendar__arrow_direction_right"
        data-disabled={false}
        data-step="1"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      />
      <div
        className="calendar__select-buttons"
      >
        <div
          className="calendar__name calendar__name_month"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="calendar__select-text"
          >
            Июнь
          </div>
          <div
            className="calendar__select-arrows"
          />
        </div>
        <div
          className="calendar__name calendar__name_year"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="calendar__select-text"
          >
            2018
          </div>
          <div
            className="calendar__select-arrows"
          />
        </div>
      </div>
    </div>
    <table
      className="calendar__layout"
    >
      <tbody>
        <tr
          className="calendar__row"
          key="row_1"
        >
          <th
            className="calendar__dayname"
            key="Пн"
          >
            Пн
          </th>
          <th
            className="calendar__dayname"
            key="Вт"
          >
            Вт
          </th>
          <th
            className="calendar__dayname"
            key="Ср"
          >
            Ср
          </th>
          <th
            className="calendar__dayname"
            key="Чт"
          >
            Чт
          </th>
          <th
            className="calendar__dayname"
            key="Пт"
          >
            Пт
          </th>
          <th
            className="calendar__dayname calendar__dayname_type_weekend"
            key="Сб"
          >
            Сб
          </th>
          <th
            className="calendar__dayname calendar__dayname_type_weekend"
            key="Вс"
          >
            Вс
          </th>
        </tr>
        <tr
          className="calendar__row"
          key="row_2"
        >
          <td
            key="day_1"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_2"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_3"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_4"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="1527811200000"
          >
            <div
              className="calendar__day"
              data-day={1527811200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              1
            </div>
          </td>
          <td
            key="1527897600000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1527897600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              2
            </div>
          </td>
          <td
            key="1527984000000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1527984000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              3
            </div>
          </td>
        </tr>
        <tr
          className="calendar__row"
          key="row_3"
        >
          <td
            key="1528070400000"
          >
            <div
              className="calendar__day"
              data-day={1528070400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              4
            </div>
          </td>
          <td
            key="1528156800000"
          >
            <div
              className="calendar__day"
              data-day={1528156800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              5
            </div>
          </td>
          <td
            key="1528243200000"
          >
            <div
              className="calendar__day"
              data-day={1528243200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              6
            </div>
          </td>
          <td
            key="1528329600000"
          >
            <div
              className="calendar__day"
              data-day={1528329600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              7
            </div>
          </td>
          <td
            key="1528416000000"
          >
            <div
              className="calendar__day"
              data-day={1528416000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              8
            </div>
          </td>
          <td
            key="1528502400000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1528502400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              9
            </div>
          </td>
          <td
            key="1528588800000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1528588800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              10
            </div>
          </td>
        </tr>
        <tr
          className="calendar__row"
          key="row_4"
        >
          <td
            key="1528675200000"
          >
            <div
              className="calendar__day"
              data-day={1528675200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              11
            </div>
          </td>
          <td
            key="1528761600000"
          >
            <div
              className="calendar__day"
              data-day={1528761600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              12
            </div>
          </td>
          <td
            key="1528848000000"
          >
            <div
              className="calendar__day"
              data-day={1528848000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              13
            </div>
          </td>
          <td
            key="1528934400000"
          >
            <div
              className="calendar__day"
              data-day={1528934400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              14
            </div>
          </td>
          <td
            key="1529020800000"
          >
            <div
              className="calendar__day"
              data-day={1529020800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              15
            </div>
          </td>
          <td
            key="1529107200000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1529107200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              16
            </div>
          </td>
          <td
            key="1529193600000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1529193600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              17
            </div>
          </td>
        </tr>
        <tr
          className="calendar__row"
          key="row_5"
        >
          <td
            key="1529280000000"
          >
            <div
              className="calendar__day"
              data-day={1529280000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              18
            </div>
          </td>
          <td
            key="1529366400000"
          >
            <div
              className="calendar__day"
              data-day={1529366400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              19
            </div>
          </td>
          <td
            key="1529452800000"
          >
            <div
              className="calendar__day"
              data-day={1529452800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              20
            </div>
          </td>
          <td
            key="1529539200000"
          >
            <div
              className="calendar__day"
              data-day={1529539200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              21
            </div>
          </td>
          <td
            key="1529625600000"
          >
            <div
              className="calendar__day"
              data-day={1529625600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              22
            </div>
          </td>
          <td
            key="1529712000000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1529712000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              23
            </div>
          </td>
          <td
            key="1529798400000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1529798400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              24
            </div>
          </td>
        </tr>
        <tr
          className="calendar__row"
          key="row_6"
        >
          <td
            key="1529884800000"
          >
            <div
              className="calendar__day"
              data-day={1529884800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              25
            </div>
          </td>
          <td
            key="1529971200000"
          >
            <div
              className="calendar__day"
              data-day={1529971200000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              26
            </div>
          </td>
          <td
            key="1530057600000"
          >
            <div
              className="calendar__day"
              data-day={1530057600000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              27
            </div>
          </td>
          <td
            key="1530144000000"
          >
            <div
              className="calendar__day"
              data-day={1530144000000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              28
            </div>
          </td>
          <td
            key="1530230400000"
          >
            <div
              className="calendar__day"
              data-day={1530230400000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              29
            </div>
          </td>
          <td
            key="1530316800000"
          >
            <div
              className="calendar__day calendar__day_type_weekend"
              data-day={1530316800000}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            >
              30
            </div>
          </td>
          <td
            key="day_7"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
        </tr>
        <tr
          className="calendar__row"
          key="row_7"
        >
          <td
            key="day_1"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_2"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_3"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_4"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_5"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_6"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
          <td
            key="day_7"
          >
            <div
              className="calendar__day calendar__day_empty"
              data-day={null}
              onClick={[Function]}
              role="gridcell"
              tabIndex={0}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</Calendar>
`;
